lib/sshutils/sftp/remote.go

```diff
@@ -37,83 +38,105 @@ func (r *remoteFS) Type() string {
 	return "remote"
 }
 
-func (r *remoteFS) Stat(ctx context.Context, path string) (os.FileInfo, error) {
+func (r *remoteFS) Glob(ctx context.Context, pattern string) ([]string, error) {
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
 
-	fi, err := r.c.Stat(path)
+	matches, err := r.c.Glob(pattern)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return matches, nil
+}
+
+func (r *remoteFS) Stat(ctx context.Context, p string) (os.FileInfo, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	fi, err := r.c.Stat(p)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	return fi, nil
 }
 
-func (r *remoteFS) ReadDir(ctx context.Context, path string) ([]os.FileInfo, error) {
+func (r *remoteFS) ReadDir(ctx context.Context, p string) ([]os.FileInfo, error) {
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
 
-	fileInfos, err := r.c.ReadDir(path)
+	fileInfos, err := r.c.ReadDir(p)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	for i := range fileInfos {
+		// if the file is a symlink, return the info of the linked file
+		if fileInfos[i].Mode().Type()&os.ModeSymlink != 0 {
+			fileInfos[i], err = r.c.Stat(path.Join(p, fileInfos[i].Name()))
```

Comment:
Can we use `filepath.Join` here and keep the argument names as `path` instead of `p` which is less descriptive?

Reply:
We can't use `filepath.Join`, but I renamed the `path` import to make the `p` parameter more descriptive